index.js

```diff
@@ -337,7 +337,7 @@
                 /* Optimal order of selectors for checking. */
                 var selector_elements = ['year', 'holiday', 'month', 'monthday', 'week', 'weekday'];
                 for (var selector_ind in selector_elements) {
-                    if (selectors[selector_elements[selector_ind]].length > 0) {
+                    if (selector_elements.hasOwnProperty(selector_ind) && selectors[selector_elements[selector_ind]].length > 0) {
```

Comment:
Ok will have a try with prettier and eventually fallback to `xo` if prettier is not mature enough. `xo` would be cool as prettier without the eslint mess so would be a very good fallback candidate.